### PR TITLE
Paym 105 authorization

### PR DESCRIPTION
**Pull request by mniesyn**
https://lodgify.atlassian.net/browse/PAYM-105

### CODE REVIEW

File: Lodgify.Payments.Stripe.Application/Services/IStripeClient.cs [hunk 1]
```diff
@@ -2,6 +2,6 @@
 
 public interface IStripeClient
 {
-    Task<Domain.Accounts.Account> CreateAccount(string country, string email, CancellationToken cancellationToken = default);
+    Task<Domain.Accounts.Account> CreateAccount(int userId, string country, string email, CancellationToken cancellationToken = default);
```
Comment (marcinmilik) [inline Lodgify.Payments.Stripe.Application/Services/IStripeClient.cs:5]: How about using naming convention for asynchronous methods with Async postfix in the name? It is usually used in Lodgify everywhere except Controller methods.

Also, I propose to never use the default for cancellationToken to force developers to always explicitly provide it.

The proposal is for this method and all others across the whole solution.

```suggestion
    Task<Domain.Accounts.Account> CreateAccountAsync(int userId, string country, string email, CancellationToken cancellationToken);
```

File: Lodgify.Payments.Stripe.Server/Controllers/AccountController.cs [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
+﻿using Lodgify.Authentication.Constants;
+using Lodgify.Payments.Stripe.Application.UseCases.CreateAccount;
 using Lodgify.Payments.Stripe.Server.Requests;
 using MediatR;
+using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 
 namespace Lodgify.Payments.Stripe.Server.Controllers;
 
+[Authorize]
```
Comment (marcinmilik) [inline Lodgify.Payments.Stripe.Server/Controllers/AccountController.cs:10]: How about making methods from this controller available only for users who are in trial or are subscribed?
Other users should not be able to create Stripe accounts. 
Most likely they should also not be able to access Stripe components (so create the session).
```suggestion
[Authorize(nameof(LodgifyAuthPolicies.AnySubscribed))]
```

File: Lodgify.Payments.Stripe.Server/Lodgify.Payments.Stripe.Server.csproj [hunk 3]
```diff
@@ -25,4 +25,8 @@
       <ProjectReference Include="..\Lodgify.Payments.Stripe.Infrastructure\Lodgify.Payments.Stripe.Infrastructure.csproj" />
     </ItemGroup>
 
+    <ItemGroup>
+      <Folder Include="Authorization\" />
+    </ItemGroup>
+
```
Comment (marcinmilik) [inline Lodgify.Payments.Stripe.Server/Lodgify.Payments.Stripe.Server.csproj:31]:
```suggestion
```

Is this needed?

File: Lodgify.Payments.Stripe.Server/Controllers/AccountController.cs (hunk 2 above)
Comment (mniesyn) [reply Lodgify.Payments.Stripe.Server/Controllers/AccountController.cs:10]: The same for account session?

**Review by marcinmilik — APPROVED** [review]

**Pull request merged**